test/services/smart-action-field-validator.test.js

```diff
@@ -50,73 +54,171 @@ describe('services > smart-action-field-validator', () => {
+        expect(() => smartActionFieldValidator.validateField(field, actionName)).toThrow(`isRequired of "${field.field}" on the smart action "${actionName}" must be a boolean.`);
       });
 
       it('should throw if field.isReadOnly is not a boolean', () => {
         expect.assertions(1);
 
+        const actionName = 'actionTest';
         const field = generateField();
         field.isReadOnly = 1;
 
-        expect(() => smartActionFieldValidator.validateField(field)).toThrow(`isReadOnly of "${field.field}" must be a boolean.`);
+        expect(() => smartActionFieldValidator.validateField(field, actionName)).toThrow(`isReadOnly of "${field.field}" on the smart action "${actionName}" must be a boolean.`);
       });
 
       it('should throw if field.reference is not a string', () => {
         expect.assertions(1);
 
+        const actionName = 'actionTest';
         const field = generateField();
         field.reference = 1;
 
-        expect(() => smartActionFieldValidator.validateField(field)).toThrow(`reference of "${field.field}" must be a string.`);
+        expect(() => smartActionFieldValidator.validateField(field, actionName)).toThrow(`reference of "${field.field}" on the smart action "${actionName}" must be a string.`);
       });
 
       it('should throw if field.type is not a valid type', () => {
         expect.assertions(1);
 
+        const actionName = 'actionTest';
         const field = generateField();
         field.type = 1;
 
-        expect(() => smartActionFieldValidator.validateField(field)).toThrow(`type of "${field.field}" must be a valid type. See the documentation for more information. https://docs.forestadmin.com/documentation/reference-guide/fields/create-and-manage-smart-fields#available-field-options`);
+        expect(() => smartActionFieldValidator.validateField(field, actionName)).toThrow(`type of "${field.field}" on the smart action "${actionName}" must be a valid type. See the documentation for more information. https://docs.forestadmin.com/documentation/reference-guide/fields/create-and-manage-smart-fields#available-field-options`);
       });
     });
   });
+
+  describe('validateFieldChangeHook', () => {
+    it('should not throw if field does not have defined hook', () => {
+      expect.assertions(1);
+
+      const field = { field: 'test' };
+      const actionName = 'actionTest';
+      const hooks = [];
+
+      expect(() => smartActionFieldValidator.validateFieldChangeHook(field, actionName, hooks))
+        .not.toThrow();
+    });
+
+    it('should not throw if field have defined hook', () => {
+      expect.assertions(1);
+
+      const field = { field: 'test', hook: 'onChange' };
+      const actionName = 'actionTest';
+      const hooks = {
+        onChange: () => {},
+      };
+
+      expect(() => smartActionFieldValidator.validateFieldChangeHook(field, actionName, hooks))
+        .not.toThrow();
+    });
+
+    it('should throw if field have not defined hook', () => {
```

Comment:
[Opinion] Labels for these tests cases aren't obvious to me

`should not throw if field does not have defined hook` => `when the field does not use the change hook feature`.`it should do nothing`
`should not throw if field have defined hook` => `when the field uses the hook feature`.`when the hook is correctly defined`.`it should do nothing`
`should throw if field have not defined hook` => `when the field uses the hook feature`.`when the field hook does not exist`.`it should throw an error` 

Would be clearer to me

Reply:
😍